### PR TITLE
Add AZURE_REGION to azure jobs

### LINKED ISSUE #1587: Azure jobs will soon require AZURE_REGION set
the create script must now have a AZURE_REGION set. I'll post a PR to add the correct variable.

### PR DESCRIPTION
**Pull request by asalkeld**
closes #1587

### CODE REVIEW

**Comment by asalkeld** [comment]
cc @jim-minter

**Comment by asalkeld** [comment]
this is needed for: https://github.com/openshift/openshift-azure/pull/488

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure-conformance.yaml [hunk 1]
```diff
@@ -219,6 +220,7 @@ objects:
           export DNS_DOMAIN=osadev.cloud
           export DNS_RESOURCEGROUP=dns
           export KUBECONFIG=/tmp/shared/_data/_out/admin.kubeconfig
+          export AZURE_REGION=eastus
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure-conformance.yaml]: I don't think we need this when deleting.

File: ci-operator/templates/openshift/openshift-azure/build-azure-image.yaml [hunk 2]
```diff
@@ -307,6 +308,7 @@ objects:
           export DNS_DOMAIN=osadev.cloud
           export DNS_RESOURCEGROUP=dns
           export KUBECONFIG=/tmp/shared/_data/_out/admin.kubeconfig
+          export AZURE_REGION=eastus
```
[the remark this answers is not in the document]
Comment (asalkeld) [reply ci-operator/templates/openshift/openshift-azure/build-azure-image.yaml]: ok, thanks for reviewing. hopefully this time it's all good

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure-upgrade.yaml [hunk 3]
```diff
@@ -107,6 +107,7 @@ objects:
         # aad integration configuration - we dont test aad so populate as dummy
         export AZURE_AAD_CLIENT_ID=$AZURE_CLIENT_ID
         export AZURE_AAD_CLIENT_SECRET=$AZURE_CLIENT_SECRET
+        export AZURE_REGION=eastus
```
Comment (kargakis) [inline ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure-upgrade.yaml]: Sorry for reviewing these one by one but upgrade.sh also does not seem to use the location. Rest lgtm

**Comment by kargakis** [comment]
/lgtm

will fix the redundant env in upgrade in a follow-up

File: ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure-upgrade.yaml (hunk 3 above)
Comment (kargakis) [reply ci-operator/templates/openshift/openshift-azure/cluster-launch-e2e-azure-upgrade.yaml]: https://github.com/openshift/release/pull/1611

**Pull request merged**